CHANGELOG.md

```diff
@@ -1,27 +1,222 @@
+- Upgraded preference settings to a top-level feature accessible directly through [Settings], new configuration options include:
+    - Whether to enable end-to-end tracing during SQL execution.
+    - Whether to continue SQL execution upon encountering errors.
+    - Customization of editor theme and font size.
+    - Configuration of editor shortcuts for running SQL commands.
+    - Setting the default workspace.
+    - Whether to enable user behavior analytics.
+    - Desktop version now supports memory size configuration through JVM parameters.
+- Added database availability indicators; the database list under projects will now show unavailable statuses and reasons.
+- Improved the initiation interaction for tickets:
+    - Support for initiating various tickets directly from the database in the resource tree.
+    - Commonly used tickets (mock data, database changes, data archiving, data cleaning, database permission application) support reinitiation with editable task parameters.
+- Enhanced database change processes to detect index change statements and automatically adjust the timeout setting (120h) to prevent index change statement execution failure due to timeout.
+- Desktop version personal settings now support custom JVM configuration with memory usage control to within 1 GB.
+- Desktop version supports exporting data files larger than 4 GB.
+- Optimized the en-US language wording of the product.
+
+### Bug fixes
+
+Connection Session
+
+- Sessions failing to remove after expiration, causing potential leaks [#2125](https://github.com/oceanbase/odc/pull/2125)
+- References to expired connection sessions failed to be promptly cleared, leading to resource leaks, which may cause increased memory consumption [#1914](https://github.com/oceanbase/odc/pull/1914)
+- Under high-frequency usage scenarios, executing SQL or viewing table data may encounter issues where the interface becomes unresponsive [#1797](https://github.com/oceanbase/odc/pull/1797)
+- Intermittent 404 errors when attempting to open SQL console [#1809](https://github.com/oceanbase/odc/pull/1809)
+
+SQL Execution
+
+- Commit/rollback button state not in sync with actual transaction status in OceanBase 4.2 [#2097](https://github.com/oceanbase/odc/pull/2097)
+- SQL statements with single-line comments unable to execute [#2085](https://github.com/oceanbase/odc/pull/2085)
+- Incorrect offsets for last SQL command without a trailing delimiter [#1970](https://github.com/oceanbase/odc/pull/1970)
+- Incompatibility of anonymous block execution module with Oracle 11g [#1759](https://github.com/oceanbase/odc/pull/1759)
+- DBMS output failed to output spaces [#1051](https://github.com/oceanbase/odc/issues/1970)
+- `#` and `$` character disappeared after SQL window formatted the code [#1490](https://github.com/oceanbase/odc/issues/1490)
+- Auto-complete in SQL window not available for MySQL datasource [#1718](https://github.com/oceanbase/odc/issues/1718)
```

Comment:
- In OceanBase v4.2, the status of the submit/rollback button is not synchronized with the actual transaction status.
- SQL statements with single-line comments fail to execute.
- The absence of a delimiter at the end of the last SQL command leads to incorrect offset calculation.
- Incorrect output in DBMS execution, spaces are not fully output.
- `#` and `$` disappear after SQL window formatting.
- Auto-complete is not available in the SQL window for the MySQL data source.